custom_components/rpi_gpio/switch.py

```diff
@@ -1,160 +1,114 @@
+    should_poll = False
+
+    def __init__(self, hub, name, port, unique_id, active_low, bias, drive, persistent):
+        _LOGGER.debug(f"GPIODSwitch init: {port} - {name} - {unique_id} - active_low: {active_low} - bias: {bias} - drive: {drive}")
+        self._hub = hub
+        self.name = name
+        self.unique_id = unique_id
         self._port = port
-        self._invert_logic = invert_logic
-        self._state = False
-        setup_output(self._port)
-        if not skip_reset:
-            write_output(self._port, 1 if self._invert_logic else 0)
+        self._attr_unique_id = unique_id
+        self._active_low = active_low
+        self._bias = bias
+        self._drive_mode = drive
+        self._persistent = persistent
 
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if device is on."""
-        return self._state
+    async def async_added_to_hass(self) -> None:
+        """Call when the switch is added to hass."""
+        await super().async_added_to_hass()
+        state = await self.async_get_last_state()
+        if not state or not self._persistent:
+            self.is_on = False
+        else: 
+            _LOGGER.debug(f"GPIODSwitch async_added_to_has initial port: {self._port} persistent: {self._persistent} state: {state.state}")
+            self.is_on = True if state.state == STATE_ON else False
+        self._hub.add_switch(self, self._port, self._active_low, self._bias, self._drive_mode)
+        self.async_write_ha_state()
 
     async def async_turn_on(self, **kwargs: Any) -> None:
-        """Turn the device on."""
-        write_output(self._port, 0 if self._invert_logic else 1)
-        self._state = True
+        self._hub.turn_on(self._port)
+        self.is_on = True
         self.async_write_ha_state()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
-        """Turn the device off."""
-        write_output(self._port, 1 if self._invert_logic else 0)
-        self._state = False
+        self._hub.turn_off(self._port)
+        self.is_on = False
         self.async_write_ha_state()
 
-
-class PersistentRPiGPIOSwitch(RPiGPIOSwitch, RestoreEntity):
-    """Representation of a persistent Raspberry Pi GPIO."""
-
-    def __init__(self, name, port, invert_logic, unique_id=None):
-        """Initialize the pin."""
-        super().__init__(name, port, invert_logic, unique_id, True)
-
-    async def async_added_to_hass(self) -> None:
-        """Call when the switch is added to hass."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_state()
-        if not state:
-            return
-        self._state = True if state.state == STATE_ON else False
-        if self._state:
-            await self.async_turn_on()
-        else:
-            await self.async_turn_off()
\ No newline at end of file
+    def update(self):
+        self.is_on = self._hub.update(self._port)
+        self.schedule_update_ha_state(False)
```

Comment:
```suggestion
```
This should cause update to happen twice, please verify if it is really needed

Reply:
I didn't notice something happens twice. @thecode, why do you suspect that? Maybe I'm missing something.